### PR TITLE
[Build] Tweak BWC tasks caching

### PR DESCRIPTION
**Pull request by breskeby**
We do not cache BWC distro creation which potentially can safe significant time in intake / pr builds/ 

Testing on checkPart4 in PR we see avoidance of ~6 minutes: https://gradle-enterprise.elastic.co/s/t6agsebb2zbjm/timeline?details=gehydmeqowjok&project=:distribution:bwc:minor

### CODE REVIEW

**Comment by breskeby** [comment]
Fixes more non cacheable tasks due to absolute path issues found here: https://gradle-enterprise.elastic.co/c/sbqckiuokaeim/63gegjth2anws/task-inputs?expanded=WyJ0Nm03eXl5M3pwdnRzLXJvb3RzcGVjJDEiLCJ0Nm03eXl5M3pwdnRzLXJvb3RTcGVjJDEtMC0wIl0

**Comment by breskeby** [comment]
Testing on checkPart4 in PR we see avoidance of about 6 minutes: https://gradle-enterprise.elastic.co/s/t6agsebb2zbjm/timeline?details=gehydmeqowjok&project=:distribution:bwc:minor

**Review by jozala — CHANGES_REQUESTED** [review]
Looks good in general, but I believe this one line is commented out by mistake.

File: build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/InternalDistributionBwcSetupPlugin.java [hunk 1]
```diff
@@ -322,13 +323,13 @@ static void createBuildBwcTask(
...
                 : projectArtifact.distFile;
-            c.getInputs().file(new File(project.getBuildDir(), "refspec"));
+            c.getInputs().file(new File(project.getBuildDir(), "refspec")).withPathSensitivity(PathSensitivity.RELATIVE);
             if (useNativeExpanded) {
                 c.getOutputs().dir(expectedOutputFile);
             } else {
                 c.getOutputs().files(expectedOutputFile);
             }
-            c.getOutputs().doNotCacheIf("BWC distribution caching is disabled for local builds", task -> buildParams.isCi() == false);
+            // c.getOutputs().doNotCacheIf("BWC distribution caching is disabled for local builds", task -> buildParams.isCi() == false);
```
[the remark this answers is not in the document]
Comment (breskeby) [reply build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/InternalDistributionBwcSetupPlugin.java]: indeed. fixed.

**Review by jozala — APPROVED** [review]

**Comment by mark-vieira** [comment]
Nice!

**Pull request merged**